### PR TITLE
Add build_and_test_linux_release presubmit test

### PR DESCRIPTION
**Pull request by liyuqian**
Also enable shell_benchmarks in postsubmit release benchmarks.

### CODE REVIEW

**Comment by liyuqian** [comment]
@chinmaygarde : the new code still fails. I wonder if `ASSERT_DEATH` works in release mode?

**Comment by liyuqian** [comment]
@chinmaygarde , I've found the issue: there's also a whitelist unit test. Although the failure is about the blacklisted flags, it now comes from the whitelist unit test instead of the blacklist unit test. Fixing whitelist unit test seems to have fixed the issue.

File: testing/run_tests.py [hunk 1]
```diff
@@ -389,7 +389,7 @@ def main():
   if 'benchmarks' in types and not IsWindows():
     RunEngineBenchmarks(build_dir, engine_filter)
 
-  if 'engine' in types or 'font-subset' in types:
+  if ('engine' in types or 'font-subset' in types) and args.variant != 'host_release':
```
Comment (liyuqian) [inline testing/run_tests.py]: @dnfield : the font-subset doesn't seem to be available in release mode. I also wonder if it's available in the profile mode. Is it debug mode only?

Comment (dnfield) [reply testing/run_tests.py]: I think only the debug mode asset is built - it should be possible to build the release mode, but not important to test it in release mode.

**Pull request merged**